src/main/java/com/urbanairship/datacube/dbharnesses/HBaseDbHarness.java

```diff
@@ -268,6 +297,44 @@ private long increment(byte[] rowKey, T op) throws IOException {
         return WithHTable.increment(pool, tableName, rowKey, cf, QUALIFIER, amount);
     }
 
+    /**
+     * @param writes map from rowkey to the operation (which had better be bytes compatible with long)
+     *
+     * @return A map from bytes to the amounts successfully added to the database.
```

Comment:
This is breaking the previous contract, which was to return the value after the increment  was applied. The inability to support the old behavior is a large part of the reason this change wasn't made earlier. I'd be +1 on either removing the onFlush capability or moving it to be Optional<> and doing reads after the write to get new values.

Reply:
It's not true that we can't retrieve the value after increment from the database. It's in the array of objects (that have to be coerced to results) for our increment operation